centreon/src/Core/HostCategory/Domain/Model/NewHostCategory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * Copyright 2005 - 2022 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+
+declare(strict_types=1);
+
+namespace Core\HostCategory\Domain\Model;
+
+use Centreon\Domain\Common\Assertion\Assertion;
+use PhpParser\Node\Expr\Cast\String_;
+
+class NewHostCategory
+{
+    public const MAX_NAME_LENGTH = 200,
+                MAX_ALIAS_LENGTH = 200,
+                MAX_COMMENT_LENGTH = 65535;
+
+    protected bool $isActivated = true;
+    protected ?string $comment = null;
+
+    public function __construct(
+        protected string $name,
+        protected string $alias
+    ) {
+        Assertion::maxLength($name, self::MAX_NAME_LENGTH, 'HostCategory::name');
+        Assertion::notEmpty($name, 'HostCategory::name');
+        Assertion::maxLength($alias, self::MAX_ALIAS_LENGTH, 'HostCategory::alias');
+        Assertion::notEmpty($alias, 'HostCategory::alias');
```

Comment:
We must avoid `notEmpty` assertions for strings : problem of the hypothetical  `'0'`.